### PR TITLE
[Core] Clean up TaxCalculationStrategy

### PR DESCRIPTION
**Pull request by pamil**

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Taxation/Strategy/TaxCalculationStrategy.php [hunk 1]
```diff
@@ -12,19 +12,79 @@
...
+    /**
      * {@inheritdoc}
      */
     public function supports(OrderInterface $order, ZoneInterface $zone)
     {
-        return $order->getChannel()->getTaxCalculationStrategy() === $this->type;
+        $channel = $order->getChannel();
+
+        /** @var ChannelInterface $channel */
+        Assert::isInstanceOf($channel, ChannelInterface::class);
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Taxation/Strategy/TaxCalculationStrategy.php]: How it is possible that `getChannel` method will return something different then instance of `ChannelInterface`? Maybe `Assert::notNull` would be a better solution.

**Comment by pjedrzejewski** [comment]
Thank you Kamil! :)

File: src/Sylius/Bundle/CoreBundle/Taxation/Strategy/TaxCalculationStrategy.php (hunk 1 above)
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Taxation/Strategy/TaxCalculationStrategy.php]: Order should always be assigned to channel.

Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Taxation/Strategy/TaxCalculationStrategy.php]: We are using `getTaxCalculationStrategy` which exists only in Core's `ChannelInterface`.

**Pull request merged**